pkg/v2/groupsync/diff.go

```diff
@@ -53,6 +53,8 @@ func Compare(before *prop.Resource, after *prop.Resource) *Diff {
 	for k := range afterIds {
 		if _, ok := beforeIds[k]; !ok {
 			diff.addJoined(k)
+		} else {
+			diff.addStayed(k)
```

Comment:
Can we check whether there's a difference in old and new name of the group (and any other relevant properties - though I can't think of anything else)? Only if the name has changed, we should increment stayed count. This will ensure that we don't make unnecessary changes to existing members if the update is a simple membership change or if there's no change at all (and the idp sent us a group message for no good reason, it's not expected but definitely possible).

Reply:
Got it, I updated the Diff structure to continue always to compute the stayed users, but to also indicate whether relevant properties have changed, I believe this makes more sense instead of not incrementing the stayed count.